coalib/collecting/Importers.py

```diff
@@ -1,34 +1,30 @@
 from contextlib import ExitStack
+from importlib.util import spec_from_file_location
 import inspect
 import os
 import platform
-import sys
 
 from coala_utils.ContextManagers import suppress_stdout
 from coala_utils.decorators import arguments_to_lists, yield_once
 
+# in Python > 3.4 this can be imported from importlib.utils
+from coalib.misc.Compatibility import module_from_spec
+
 
 def _import_module(file_path):
     if not os.path.exists(file_path):
-        raise ImportError
+        raise ImportError("can's import non-existing file %s"
+                          % repr(file_path))
 
     module_name = os.path.splitext(os.path.basename(file_path))[0]
-    module_dir = os.path.dirname(file_path)
-
-    if module_dir not in sys.path:
-        sys.path.insert(0, module_dir)
```

Comment:
You still need to do this, otherwise `spec.loader.exec_module` can't load the module.